components/omega/doc/design/ShallowWaterOmega0.md

```diff
@@ -0,0 +1,239 @@
+
+### 2.6 Requirement: Omega-0 will run on multi-node with domain decomposition
+
+This is with MPI and halo communication, as described in framework design documents. Results must be bit-for-bit identical across different number of partitions. This may be demonstrated with the ''QU240 partition test'' in Polaris.
+
+### 2.7 Requirement: Correct convergence rates of operators and exact solutions
+
+Each operator will be tested individually for the proper convergence rate: divergence, gradient, curl interpolated to cell centers, and tangential velocity are all second order; curl at vertices is first order. The details of the convergence tests are explained in section 5.1 below.
+
+### 2.8 Requirement: Conservation of volume and tracer
+
+The total volume of the domain should be conserved to machine precision in the absence of surface volume fluxes. Likewise, the total tracer amount should be conserved to machine precision without surface fluxes. A simple test is to initialize an inactive test tracer with a uniform value of 1.0, and it should remain 1.0 throughout the domain.
+
+### 2.9 Requirement: Performance will be at least as good or better than MPAS-Ocean
+
+Performance will be assessed with the following metrics:
+
+1. Single CPU throughput
+1. Parallel CPU scalability to high node counts
+1. Single GPU throughput
+
+The first two items will be compared to MPAS-Ocean with the same test configuration. For Omega-0, these may be tested first with the layered shallow water equations (momentum and thickness only) and be compared directly to the results in [Bishnu et al. 2023a](https://egusphere.copernicus.org/preprints/2023/egusphere-2023-57), using the inertia-gravity wave test case described in Section 6.2 below. After tracer transport capabilities are added, performance may be compared against MPAS-Ocean with test cases using active tracers in addition to momentum and thickness.
+
+For these comparisons, MPAS-Ocean will be set up with the identical terms and functionality as Omega-0. This means that vertical advection, vertical mixing, and all parameterizations will be disabled. Comparisons with these terms will be made with Omega-1 and higher and will be described in future design documents.
+
+### 2.10 Requirement: Full-node GPU throughput will be comparable or better than full-node CPU throughput
+
+For GPU throughput, comparisons should be made between full-node CPU throughput and full-node GPU throughput. For example, [Perlmutter at NERSC](https://docs.nersc.gov/systems/perlmutter/architecture/) has nodes with 64 and 128 CPU cores (AMD EPYC 7763) and 4 GPUs (NVIDIA A100). We expect the full-node GPU throughput to be at least as good as the full-node CPU throughput, and potentially a factor of four higher. These numbers depend on the performance specifications of the particular hardware.
```

Comment:
@philipwjones and @sarats, is this a reasonable requirement for GPU performance for Omega-0? I'm happy to alter or reword.

Reply:
It's fine for a requirement. Aspirations are for higher, but that's mentioned too.